### ISSUE TITLE
Gradient computation modified by an in-place operation 

### ISSUE THREAD

**Issue by lexili24** [hbutsuak95/monolayout#6]
Hi! Thank you for your work, it is truly inspiring and amusing to play with. I have a question regarding details on training with discriminator networks. I run into an issue of `RuntimeError:` one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [2, 16, 3, 3]] is at version 2; expected version 1 instead." It seems like that error is only present when I add discriminator networks, I am trying to feed in the discriminator networks with the output of decoder network and a ground truth right here. I tried to clone a bunch of variables like `fake` and `valid`, as well as outputs from dynamic & static decoders and it is still here. If there is any suggestion that would be great. Thanks ahead.

**Comment by lexili24** [comment]
with `anomaly_detection` flag on I am able to further trace and it seems to go back at the decoder network. 
`
Warning: Error detected in CudnnConvolutionBackward. Traceback of forward call that caused the error:
  File "train_mono.py", line 209, in <module>
    outputs['static'] = models['static'](encoded_features)
  File "../module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "../model.py", line 377, in forward
    x = self.convs["topview"](x)
  File "../module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "../model.py", line 328, in forward
    out = self.conv(out)
  File "../.local/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "../.local/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 349, in forward
    return self._conv_forward(input, self.weight)
  File "../.local/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 346, in _conv_forward
    self.padding, self.dilation, self.groups)
 (print_stack at /pytorch/torch/csrc/autograd/python_anomaly_mode.cpp:60)
`

**Comment by krrish94** [comment]
Superficially, appears more of an error related to pytorch version? Are you using 1.4/1.5 by any chance? Much of our tests were on pytorch 1.3 and 1.1.

**Comment by lexili24** [comment]
Thanks for the feedback! I was able to resolve this issue by making sure both `.step()` run after both `.backward()`.
